roachtest: psycopg failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19654287?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19654287?buildTab=artifacts#/psycopg) on master @ [508919efdd82778cbf718f917e13ceca89020d39](https://github.com/cockroachdb/cockroach/commits/508919efdd82778cbf718f917e13ceca89020d39):


```
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.TEXT-b] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50513

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19654276?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19654276?buildTab=artifacts#/psycopg) on master @ [508919efdd82778cbf718f917e13ceca89020d39](https://github.com/cockroachdb/cockroach/commits/508919efdd82778cbf718f917e13ceca89020d39):


```
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19654276/psycopg/1746775113946/1746776005009)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19661492?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19661492?buildTab=artifacts#/psycopg) on master @ [58f7422be6001c29dfca0311aa1d3fa723cd96f1](https://github.com/cockroachdb/cockroach/commits/58f7422be6001c29dfca0311aa1d3fa723cd96f1):


```
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-t] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicLeases=leader</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19661492/psycopg/1746865855000/1746866521382)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19661503?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19661503?buildTab=artifacts#/psycopg) on master @ [58f7422be6001c29dfca0311aa1d3fa723cd96f1](https://github.com/cockroachdb/cockroach/commits/58f7422be6001c29dfca0311aa1d3fa723cd96f1):


```
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.BINARY-s] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19664285?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19664285?buildTab=artifacts#/psycopg) on master @ [6806299943a595425030f0ad3c54ccc680c89842](https://github.com/cockroachdb/cockroach/commits/6806299943a595425030f0ad3c54ccc680c89842):


```
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-namedtuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-t] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19666729?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19666729?buildTab=artifacts#/psycopg) on master @ [6806299943a595425030f0ad3c54ccc680c89842](https://github.com/cockroachdb/cockroach/commits/6806299943a595425030f0ad3c54ccc680c89842):


```
--- FAIL: tests.test_cursor_client.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19666718?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19666718?buildTab=artifacts#/psycopg) on master @ [6806299943a595425030f0ad3c54ccc680c89842](https://github.com/cockroachdb/cockroach/commits/6806299943a595425030f0ad3c54ccc680c89842):


```
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19666718/psycopg/1747032436467/1747033400439)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Next Steps: Try a git bisect to find out when this first started occurring.  To speed up the bisect, we can run only one of the failing tests to find our target commit.

Example of 1 of the failures:
```
tests/test_cursor_raw_async.py:114: AssertionError
________________ test_leak[asyncio-dict_row-iter-Format.TEXT-s] ________________

aconn_cls = <class 'psycopg.crdb.AsyncCrdbConnection'>
dsn = 'host=localhost port=26257 user=roachprod password=cockroachdb dbname=defaultdb'
faker = <tests.fix_faker.Faker object at 0x71a54ede3490>
fmt = <PyFormat.AUTO: 's'>, fmt_out = <Format.TEXT: 0>, fetch = 'iter'
row_factory = <function dict_row at 0x71a5513905e0>
gc = <tests.fix_gc.GCFixture object at 0x71a54ee290a0>

    @pytest.mark.slow
    @pytest.mark.parametrize("fmt", PyFormat)
    @pytest.mark.parametrize("fmt_out", pq.Format)
    @pytest.mark.parametrize("fetch", ["one", "many", "all", "iter"])
    @pytest.mark.parametrize("row_factory", ["tuple_row", "dict_row", "namedtuple_row"])
    async def test_leak(aconn_cls, dsn, faker, fmt, fmt_out, fetch, row_factory, gc):
        faker.format = fmt
        faker.choose_schema(ncols=5)
        faker.make_records(10)
        row_factory = getattr(rows, row_factory)
    
        async def work():
            async with await aconn_cls.connect(dsn) as aconn:
                async with aconn.transaction(force_rollback=True):
                    async with aconn.cursor(binary=fmt_out, row_factory=row_factory) as cur:
                        await cur.execute(faker.drop_stmt)
                        await cur.execute(faker.create_stmt)
                        async with faker.find_insert_problem_async(aconn):
                            await cur.executemany(faker.insert_stmt, faker.records)
                        await cur.execute(ph(cur, faker.select_stmt))
    
                        if fetch == "one":
                            while True:
                                tmp = await cur.fetchone()
                                if tmp is None:
                                    break
                        elif fetch == "many":
                            while True:
                                tmp = await cur.fetchmany(3)
                                if not tmp:
                                    break
                        elif fetch == "all":
                            await cur.fetchall()
                        elif fetch == "iter":
                            async for rec in cur:
                                pass
    
        n = []
        gc.collect()
        for i in range(3):
            await work()
            gc.collect()
            n.append(gc.count())
>       assert n[0] == n[1] == n[2], f"objects leaked: {n[1] - n[0]}, {n[2] - n[1]}"
E       AssertionError: objects leaked: 8, 8
E       assert 200476 == 200484
```

`objects leaked` is a keyword that can be searched to find other similar failing tests.

Note: Checkout psycopg repo and find instructions for how to run just one of the failing tests there.

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19685367?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19685367?buildTab=artifacts#/psycopg) on master @ [f903e26dff81dacc1e67662cfa9652b49d82a9c2](https://github.com/cockroachdb/cockroach/commits/f903e26dff81dacc1e67662cfa9652b49d82a9c2):


```
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-b] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19685356?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19685356?buildTab=artifacts#/psycopg) on master @ [f903e26dff81dacc1e67662cfa9652b49d82a9c2](https://github.com/cockroachdb/cockroach/commits/f903e26dff81dacc1e67662cfa9652b49d82a9c2):


```
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-t] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19685356/psycopg/1747206583920/1747207686117)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19694983?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19694983?buildTab=artifacts#/psycopg) on master @ [dc17e6481cb92b253cbb54c1bb7608c0e46fb2e5](https://github.com/cockroachdb/cockroach/commits/dc17e6481cb92b253cbb54c1bb7608c0e46fb2e5):


```
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-namedtuple_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-iter-Format.BINARY-t] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19694972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19694972?buildTab=artifacts#/psycopg) on master @ [dc17e6481cb92b253cbb54c1bb7608c0e46fb2e5](https://github.com/cockroachdb/cockroach/commits/dc17e6481cb92b253cbb54c1bb7608c0e46fb2e5):


```
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-iter-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-tuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.TEXT-t] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19694972/psycopg/1747312036048/1747312915874)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19705291?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/19705291?buildTab=artifacts#/psycopg) on master @ [1a847b18290d439c62ff739ebdae00c851d5442e](https://github.com/cockroachdb/cockroach/commits/1a847b18290d439c62ff739ebdae00c851d5442e):


```
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client.test_leak[asyncio-dict_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-many] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19705280?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/19705280?buildTab=artifacts#/psycopg) on master @ [1a847b18290d439c62ff739ebdae00c851d5442e](https://github.com/cockroachdb/cockroach/commits/1a847b18290d439c62ff739ebdae00c851d5442e):


```
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-tuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-one-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-all] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-one-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[dict_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-tuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-tuple_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-dict_row-one] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-one-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-iter-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-dict_row-all-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-one-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-many-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-all-Format.BINARY-t] - unknown (unexpected)
--- FAIL: tests.test_cursor.test_leak[tuple_row-all-Format.TEXT-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.BINARY-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-namedtuple_row-all-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_client_async.test_leak[asyncio-namedtuple_row-iter] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-dict_row-many-Format.BINARY-b] - unknown (unexpected)
--- FAIL: tests.test_cursor_async.test_leak[asyncio-tuple_row-many-Format.TEXT-t] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw_async.test_leak[asyncio-dict_row-iter-Format.TEXT-s] - unknown (unexpected)
--- FAIL: tests.test_cursor_raw.test_leak[asyncio-namedtuple_row-many-Format.TEXT-s] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-19705280/psycopg/1747378791196/1747379646003)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146429 roachtest: psycopg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2]
- #146427 roachtest: psycopg failed [C-test-failure O-roachtest O-robot P-1 T-sql-foundations branch-release-25.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/psycopg.go
+++ b/pkg/cmd/roachtest/tests/psycopg.go
@@ -19,7 +19,7 @@ import (
 )
 
 var psycopgReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<point>\d+)(?:\.(?P<subpoint>\d+))?)?)?$`)
-var supportedPsycopgTag = "3.2.6"
+var supportedPsycopgTag = "3.2.8"
 
 // This test runs psycopg full test suite against a single cockroach node.
 func registerPsycopg(r registry.Registry) {

